src/audio.c

```diff
@@ -172,8 +172,20 @@ struct aurx {
 		uint64_t n_discard;
 	} stats;
 
-	struct tmr tmr;       /**< Timer for audio decoding        */
-	volatile int32_t wcnt;
+	volatile int32_t wcnt;       /**< Write handler call count         */
+
+	union {
+#ifdef HAVE_PTHREAD
+		struct {
+			pthread_t tid;
+			bool run;
+			pthread_cond_t cond;
+			pthread_mutex_t mutex;
+		} thr;                /**< Audio decode thread             */
+#else
+		struct tmr tmr;       /**< Timer for audio decoding        */
+#endif
+	} u;
```

Comment:
I think with ifdef/else there is only one union member at time, so the union looks a bit needless?

Reply:
Yes. I'll remove this. Same for the struct autx? I ask already 3 days ago.